### PR TITLE
Implement frame timing callbacks in Skwasm.

### PR DESCRIPTION
**Pull request by eyebrowsoffire**
Fixes https://github.com/flutter/flutter/issues/140429

Some notes here:
* Refactored the frame timing systems so that we can deal with asynchronous rendering.
* Consolidated rendering of multiple pictures in skwasm into a single call, so that the rasterization can be properly measured.
* Pulled the frame timings tests into the `ui` test suite so that they run on all renderers (including skwasm).

### CODE REVIEW

File: lib/web_ui/lib/frame_timings.dart
Comment (mdebbar) [inline lib/web_ui/lib/frame_timings.dart:1]: I strongly recommend that we keep these files as similar as possible to their `lib/ui/` counterparts to keep maintenance of the `dart:ui` API surface easy across platforms.

Comment (yjbanov) [reply lib/web_ui/lib/frame_timings.dart:1]: I agree. We deviated before, which was difficult to maintain.

**Review by yjbanov — APPROVED** [review]
lgtm

File: lib/web_ui/lib/src/engine/skwasm/skwasm_impl/surface.dart [hunk 1]
```diff
@@ -12,6 +12,17 @@ import 'package:ui/src/engine.dart';
...
 import 'package:ui/ui.dart' as ui;
 
+@JS()
+@staticInterop
+@anonymous
+class RasterResult {}
+
+extension RasterResultExtension on RasterResult {
+  external JSNumber get rasterStart;
+  external JSNumber get rasterEnd;
```
[the remark this answers is not in the document]
Comment (eyebrowsoffire) [reply lib/web_ui/lib/src/engine/skwasm/skwasm_impl/surface.dart:22]: It's not actually micros, it's milliseconds as a double (what `performance.now` returns). I guess I should rename them to match that though, or at least put a comment in.

File: lib/web_ui/lib/frame_timings.dart
Comment (eyebrowsoffire) [reply lib/web_ui/lib/frame_timings.dart:1]: Kind of makes me sad but okay.

**Review by mdebbar — APPROVED** [review]
Overall LGTM, but I'm also leaning on Yegor's LGTM for the rendering bits that I'm not too familiar with :)

**Comment by goderbauer** [comment]
Reason for revert: Fails in device lab, see https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8755350727803344657/+/u/run_web_benchmarks_skwasm/stdout

**Pull request merged**